src/sumcheck/sumcheck_single.rs

```diff
@@ -152,95 +279,29 @@ where
 /// The linear coefficient c_1 is derived by the verifier using the sum constraint:
 /// h(0) + h(1) = claimed_sum, so h(1) = claimed_sum - c_0.
 #[instrument(skip_all, level = "debug")]
-fn compute_sumcheck_coefficients<F: Field, EF: ExtensionField<F>>(
-    evals: &EvaluationsList<F>,
-    weights: &EvaluationsList<EF>,
-) -> (EF, EF) {
-    assert!(evals.num_variables() >= 1);
+fn compute_sumcheck_coefficients<A, B>(evals: &[A], weights: &[B]) -> (B, B)
+where
+    A: Copy + Send + Sync + Algebra<A>,
+    B: Copy + Send + Sync + Algebra<A> + Algebra<B>,
```

Comment:
Requiring something be an algebra over itself is a bit meaningless. I think the "right" trait bounds here are:
```suggestion
    A: Copy + Send + Sync + PrimeCharacteristicRing,
    B: Copy + Send + Sync + Algebra<A>,
```
Note that being an algebra implies you are a `PrimeCharacteristicRing`.